File: vita/src/main/java/vita/ep/m5/finalized/Neg.java
package vita.ep.m5.finalized;

import vita.ep.m5.Exp;

public class Neg extends Factory implements vita.ep.m5.Neg<Visitor> {

    protected Exp<Visitor> inner;

    public Neg(Exp<Visitor> _inner) {
        this.inner = _inner;
    }

    public Exp<Visitor> getInner() {
        return this.inner;
    }

    public void accept(Visitor visitor) {
        visitor.visit(this);
    }

    public Exp<Visitor> convert(vita.ep.Exp<Visitor> other) {
        Visitor visitor = new Visitor();
        other.accept(visitor);
        return visitor.getResult();
    }
}

File: coco_csharp/CoCo/EP/M3/Finalized/Neg.cs
namespace EP.M3.Finalized {
    public class Neg : EP.M2.Finalized.Exp, EP.M3.Neg<EP.M2.Finalized.Exp>, Factory {
        private EP.Exp<EP.M2.Finalized.Exp> inner;

        public Neg(EP.Exp<EP.M2.Finalized.Exp> _inner) {
            this.inner = _inner;
        }

        public EP.Exp<EP.M2.Finalized.Exp> getInner() { return this.inner; }
    }
}